rcl/test/rcl/test_events.cpp

```diff
@@ -555,3 +558,130 @@ TEST_F(CLASSNAME(TestEventFixture, RMW_IMPLEMENTATION), test_pubsub_liveliness_k
   // clean up
   tear_down_publisher_subscriber();
 }
+
+/*
+ * Basic test of publisher and subscriber incompatible qos callback events.
+ * Only implemented in opensplice at the moment.
+ */
+TEST_F(CLASSNAME(TestEventFixture, RMW_IMPLEMENTATION), test_pubsub_incompatible_qos)
```

Comment:
This test is particularly long because it is checking for every possible qos mismatch. Only the `qos_profiles` passed need to be changed. The core portion of the test can be moved into another function and then each qos policy incompatibility can be moved into a test of its own; if this test is difficult to understand.